src/newgrf_text.cpp

```diff
@@ -561,27 +560,22 @@ StringID AddGRFString(uint32_t grfid, uint16_t stringid, byte langid_to_add, boo
 		}
 	}
 
-	uint id;
-	for (id = 0; id < _num_grf_texts; id++) {
-		if (_grf_text[id].grfid == grfid && _grf_text[id].stringid == stringid) {
-			break;
-		}
+	auto it = std::find_if(std::begin(_grf_text), std::end(_grf_text), [&grfid, &stringid](const GRFTextEntry &grf_text) { return grf_text.grfid == grfid && grf_text.stringid == stringid; });
+	if (it == std::end(_grf_text)) {
+		/* Too many strings allocated, return empty */
+		if (_grf_text.size() == TAB_SIZE_NEWGRF) return STR_EMPTY;
+
+		/* We didn't find our stringid and grfid in the list, allocate a new id */
+		_grf_text.emplace_back();
+		it = std::end(_grf_text) - 1;
```

Comment:
```suggestion
		it = _grf_text.emplace(std::end(_grf_text));
```
Maybe ?

Reply:
Thanks, that's the construct I was trying to find but failing...